src/main/kotlin/com/fantasy/ladbe/oauth/jwt/JwtFilter.kt

```diff
@@ -0,0 +1,46 @@
+package com.fantasy.ladbe.oauth.jwt
+
+import com.fantasy.ladbe.handler.exception.Exceptions
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.web.filter.OncePerRequestFilter
+import javax.servlet.FilterChain
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+const val AUTHORIZATION_HEADER: String = "Authorization"
+
+class JwtFilter(
+    private val jwtProvider: JwtProvider
+) : OncePerRequestFilter() {
+
+    override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, chain: FilterChain) {
+        val jwt: String = resolveToken(request)
+
+        if (jwt.isNotEmpty() && jwtProvider.validateToken(jwt, request)) {
+            val authentication: Authentication = jwtProvider.parseJwt(jwt)
+            SecurityContextHolder.getContext().authentication = authentication
+        }
+
+        chain.doFilter(request, response)
+    }
+
+    /**
+     * 헤더에서 JWT 를 가져온 뒤, 유효성 검토를 위해 prefix 를 제외하는 작업을 하는 메소드
+     * param : request - f/e에 받은 요청 정보
+     * return : 헤더에 Authorization 이름으로 값이 있는지 확인한다.
+     *          값이 없다면, "" -> 빈 문자열을 반환하고 null 예외처리 발생
+     *          값이 있다면, 빈 값이 들어있는지 확인하고 Bearer 로 시작하는지 확인한다. 그리고 있다면 prefix 를 제거하고 반환
+     */
+    private fun resolveToken(request: HttpServletRequest): String {
+        val token: String = request.getHeader(AUTHORIZATION_HEADER)?.let {
+            it
+        } ?: ""
+
+        if (token.isNotEmpty() && token.startsWith("Bearer "))
+            return token.substring(7)
+
+        request.setAttribute("exception", Exceptions.JWT_NULL_TOKEN.code)
+        throw NullPointerException()
```

Comment:
```suggestion
        throw BusinessException(Exceptions.JWT_NULL_TOKEN)
```

NPE로 설정하신 이유가 있을까요? (없으시다면 저희가 쓰도록한 BusinessException...ㅎㅎ)

Reply:
@kimdg1105 
저희가 사용하려고 하는 BusinessException은 `Spring Context` 에서 사용이 되고있는 예외처리입니다.

JWT 는 `Filter` 단에서 진행이 되어 Spring Context에 들어가기 전, `Web Application` 에서 예외처리가 일어나게 됩니다.

제안하신 방식대로 진행하게 된다면, `BusinessException` 이 예외는 잡으나 (throw 로 직접 예외를 던졌기 때문입니다.), 그에대한 응답이 제대로 내려지지 않았습니다.
`ResponseEntity` 를 통해 응답을 보내려고 해도, ResponseEntity는 Spring Framework에서 제공하는 클래스이기 때문에 빈 응답만 내려오게 됩니다.

Web Application 단에서 생성된 예외를 바로 던져, Spring Context 에 도달하지 못하게 만들기 위해 NPE를 바로 던지게 해놨습니다.